src/utils/proposal.ts

```diff
@@ -97,6 +97,18 @@ export function snapshotUrl(hash: string) {
   return target.toString()
 }
 
+export function getSnapshotIdFromUrl(snapshotUrl: string) {
+  const target = new URL(snapshotUrl)
+  const hash = target.hash.slice(1)
+
+  const proposalMatch = hash.match(/\/proposal\/([^/]+)$/)
+  if (proposalMatch && proposalMatch[1]) {
+    return proposalMatch[1]
+  }
+
+  return hash.split('/').pop() || ''
+}
+
```

Comment:
I'd leave this function in the submit page if it's not being used anywhere else.